criu/net.c

```diff
@@ -1818,6 +1819,52 @@ static inline int dump_iptables(struct cr_imgset *fds)
 	return 0;
 }
 
+static inline int dump_nftables(struct cr_imgset *fds)
+{
+	int ret = -1;
+	struct cr_img *img;
+	int img_fd;
+	FILE *fp;
+	struct nft_ctx *nft;
+
+	/* we not dump nftables if nft utility isn't present */
+	if (!check_cmd_exists("nft"))
+		return 0;
+
+	nft = nft_ctx_new(NFT_CTX_DEFAULT);
+	if (!nft)
+		return -1;
+
+	img = img_from_set(fds, CR_FD_NFTABLES);
+	img_fd = dup(img_raw_fd(img));
+	if (img_fd < 0) {
+		pr_perror("dup() failed");
+		goto nft_ctx_free_out;
+	}
+
+	fp = fdopen(img_fd, "w");
+	if (!fp) {
+		pr_perror("fdopen() failed");
+		goto nft_ctx_free_out;
+	}
+
+	nft_ctx_set_output(nft, fp);
+
+	if (nft_run_cmd_from_buffer(nft, "list ruleset")) {
+		goto fp_close_out;
+	}
+
+	fflush(fp);
```

Comment:
Why we need fflush here, isn't it enough to fclose?

Reply:
my bad :(